packages/js-lib/src/enum.util.ts

```diff
@@ -1,4 +1,34 @@
-import type { NumberEnum, StringEnum } from './types.js'
+import type { AnyObject, NumberEnum, StringEnum } from './types.js'
+
+export function isNumberEnum(en: AnyObject): en is NumberEnum {
+  /*
+   * enum Foo { A = 1, B = 2 }
+   * becomes
+   * { "1": "A", "2": "B", "A": 1, "B": 2}
+   */
+
+  return Object.values(en).every(value => {
+    const isValueString = isNaN(Number(value))
+    const isValueNumber = !isValueString
+
+    if (isValueNumber) {
+      return en[value] !== undefined
+    }
+
+    if (isValueString) {
+      return !isNaN(Number(en[value]))
+    }
+  })
+}
+
+export function isStringEnum(en: AnyObject): en is StringEnum {
+  /*
+   * enum Foo { A = "K1", B = "K2" }
+   * becomes
+   * { "K1": "A", "K2": "B", "A": 1, "B": 2}
+   */
+  return Object.values(en).every(value => isNaN(Number(value)))
+}
```

Comment:
Looks good.

But if it's gonna be at the code of our "performant library code" - maybe we can go old-school and do a for-loop.
:old-man-yells-at-callback:

Reply:
Even if this runs only "once" when the schema is compiled, and not every time we validate?

Reply:
Yes, I know that it runs once.

It affects server bootstrap time, in NCB3 we easily have 100+ places with enum schemas.

In previous version of unit tests (with `isolate: true`), all schemas were recompiled for every single test.